xbmc/pvr/epg/EpgContainer.cpp

```diff
@@ -759,8 +755,17 @@ bool CPVREpgContainer::UpdateEPG(bool bOnlyPending /* = false */)
   if (bShowProgress && !bOnlyPending)
     progressHandler->DestroyProgress();
 
+  database->Lock();
   for (const auto& epg : invalidTables)
-    DeleteEpg(epg);
+  {
+    QueueDeleteEpg(epg);
+
+    size_t queryCount = database->GetDeleteQueriesCount();
+    if (queryCount > 10000)
```

Comment:
I feel we should define a constant in EpgDateBase.h and PVRDatabase.h for this magic number now that is is used all over the place.

Reply:
Done